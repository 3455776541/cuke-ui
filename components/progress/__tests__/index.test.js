import React from "react";
import assert from "power-assert";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import Progress from "../index";

describe("<Progress/>", () => {
	it("should render a <Progress/> components", () => {
		const wrapper = render(
			<div>
				<Progress percent={70}/>

				<Progress percent={70} animation={false}/>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should render five type width <Progress/> ", () => {
		const wrapper = render(
			<div>
				<Progress percent={20} type="default"/>
				<Progress percent={30} type="success"/>
				<Progress percent={40} type="info"/>
				<Progress percent={50} type="warning"/>
				<Progress percent={60} type="error"/>

				<Progress percent={20} type="default" animation={false}/>
				<Progress percent={30} type="success" animation={false}/>
				<Progress percent={40} type="info" animation={false}/>
				<Progress percent={50} type="warning" animation={false}/>
				<Progress percent={60} type="error" animation={false}/>
			</div>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should find cuke-progress classnames", () => {
		const wrapper = render(
			<div>
				<Progress percent={20} />
				<Progress percent={20} animation={false}/>
			</div>
		);
		assert(wrapper.find(".cuke-progress").length >= 1);
		assert(wrapper.find(".cuke-progress-enter").length >= 1);
		assert(wrapper.find(".cuke-progress-bg-animation").length === 1);
		assert(wrapper.find(".cuke-progress-bg").length >= 1);
	});
});
