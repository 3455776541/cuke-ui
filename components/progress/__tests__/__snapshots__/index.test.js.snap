// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress/> should render a <Progress/> components 1`] = `
<div>
  <div
    class="cuke-progress cuke-progress-default"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:70%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      70%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-default"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:70%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      70%
    </div>
  </div>
</div>
`;

exports[`<Progress/> should render five type width <Progress/>  1`] = `
<div>
  <div
    class="cuke-progress cuke-progress-default"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:20%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      20%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-success"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:30%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      30%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-info"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:40%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      40%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-warning"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:50%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      50%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-error"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:60%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      60%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-default"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:20%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      20%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-success"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:30%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      30%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-info"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:40%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      40%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-warning"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:50%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      50%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-error"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:60%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      60%
    </div>
  </div>
</div>
`;
